azure-pipelines.yml

```diff
@@ -219,6 +219,47 @@ jobs:
     parameters:
       artifact: top_earlgrey_verilator
 
+- job: "execute_verilated_tests"
+  displayName: "Execute tests on the Verilated system"
+  pool: "Default"
+  dependsOn:
+    - top_earlgrey_verilator
+    - sw_build
+  steps:
+  - bash: |
+      sudo apt-get install -y python3 python3-pip build-essential python3-setuptools libelf1 tree
+      sudo pip3 install -r python-requirements.txt
+    displayName: 'Install dependencies'
+  - template: 'ci/download-artifacts-template.yml'
+  - bash: |
+      . util/build_consts.sh
+      export VERILATED_SYSTEM_PATH="$BIN_DIR/hw/top_earlgrey/Vtop_earlgrey_verilator"
+      ci/run_verilator_pytest.sh
+    displayName: 'Execute tests'
+
+- job: "riscv_compliance_tests"
+  displayName: "Execute RISC-V compliance tests"
+  pool: "Default"
+  dependsOn:
+    - top_earlgrey_verilator
+    - execute_verilated_tests
```

Comment:
That dependency is unnecessary, isn't it?

Reply:
so i actually put this here on purpose because i did not want the riscv tests to run if the normal verilated tests fail (since those are super short and fast, and the riscv tests take a bit longer).

As i'm saying that now though, I'm guessing this does not really work since the pass / failure of a particular dependency probably is not factored in...

Reply:
i added a condition check instead of taking it out. Let me know what you think, or if this is the wrong rationale, happy to take it out also.

Reply:
Creating a dependency like that serializes steps and increases end-to-end CI latency; that's bad for developer productivity: you want to get results as quickly as possible.

Also, think of this scenario: you have compliance suite and the other tests fail for different reasons. On the first push, you see that the verilator tests fail, and fix that. Now you push again, and see that the compliance tests fail. You need to do another push to fix that. As result, you've lost ~20 min waiting for another CI run, and even more in terms of productivity as you have to context-switch between tasks.

So the general rule: only introduce dependencies if you need them for correctness. It's much cheaper (at this level) to buy machine time than developer time.

Reply:
i see what you mean, but i think it's pretty unlikely (in this specific scenario) for our sanity tests to fail and somehow riscv CI to pass though... and since the riscv suite is fairly big (likely to exceed 70 tests long term, each compiled separately and loaded), it will definitely take a ton of time.  My view was that it'd be unnecessary to run it unless we know for sure the system functions at least at a basic level.

I'll take out the dependency for now though.

Reply:
@imphil can you have a look now?